sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/util/StringUtils.scala

```diff
@@ -65,12 +65,15 @@ object StringUtils extends Logging {
     "(?s)" + out.result() // (?s) enables dotall mode, causing "." to match new lines
   }
 
-  private[this] val trueStrings = Set("t", "true", "y", "yes", "1").map(UTF8String.fromString)
-  private[this] val falseStrings = Set("f", "false", "n", "no", "0").map(UTF8String.fromString)
+  private[this] val trueStrings =
```

Comment:
Could you add a comment before this line? Something like the following.
> "true", "yes", "1", "false", "no", "0", and unique prefixes of these strings are accepted.

Otherwise, it's hard to recognize why we don't accept "o" here.